bin/build.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# Builds and uploads:
+#   - CLI to S3
+#   - Spark docker image to dockerhub
+#   - stub universe zip to S3
+#
+# Manifest config:
+#   CLI_VERSION - version label to use for CLI package
+#   SPARK_DIST_URI - where fetch spark distribution from
+#
+# ENV vars:
+#   COMMONS_TOOLS_DIR - path to dcos-commons/tools/, or empty to fetch latest release tgz
+#   DOCKER_IMAGE - "<image>:<version>", falls back to mesosphere/spark-dev:COMMIT)
+#   ghprbActualCommit / GIT_COMMIT - COMMIT value to use for DOCKER_IMAGE, if DOCKER_IMAGE isn't specified
+
+set +e
+
+BIN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+BASEDIR="${BIN_DIR}/.."
+
+configure_docker_image() {
+    if [ -z "$DOCKER_IMAGE" ]; then
+        # determine image label based on git commit:
+        if [ -n "$ghprbActualCommit" ]; then
+            # always overrides default GIT_COMMIT:
+            export GIT_COMMIT=$ghprbActualCommit
```

Comment:
I wasn't sure if it was needed to get values out of functions. Verified with a test script that it isn't. Removed.